### PR TITLE
tests for addressbook classes

### PR DESCRIPTION
**Pull request by lucc**
(only some first tests)

### CODE REVIEW

File: tests/addressbook/abook_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        [1]
+        name = me
+        email = me@example.com
+        [2]
+        name = you
+        email = you@other.domain, you@example.com
+        """
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(data)
+            path = tmp.name
```
Comment (dcbaker) [inline tests/addressbook/abook_test.py]: you should add: `self.addCleanup(os.unlink, path)` here to ensure that the temporary file is removed.

File: tests/addressbook/external_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+from alot.addressbook import external
+
+
+class TestExternalAddressbookGetContacts(unittest.TestCase):
+
+    """Some test cases for
+    alot.addressbook.external.ExternalAddressbook.get_contacts"""
+
+    regex = '(?P<name>.*)\t(?P<email>.*)'
```
Comment (dcbaker) [inline tests/addressbook/external_test.py]: this should be a a raw-string (`r'(?P...`)

File: alot/addressbook/external.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def _call_and_parse(self, commandline):
             m = re.match(self.regex, l, self.reflags)
             if m:
                 info = m.groupdict()
-                if 'email' and 'name' in info:
+                if 'email' in info and 'name' in info:
```
Comment (dcbaker) [inline alot/addressbook/external.py]: Ohh, that's a subtle bug. Nice catch.

File: tests/addressbook/external_test.py (hunk 2 above)
Comment (lucc) [inline tests/addressbook/external_test.py]: Why? Does it matter if that is a literal tab or the two byte sequence `\t`? And there are not many backslashes in that string so `r'...'` doesn't make it any more readable.

Comment (pazz) [inline tests/addressbook/external_test.py]: I'm starting to believe in tests after all :) nice one.

File: alot/addressbook/external.py (hunk 3 above)
Comment (pazz) [inline alot/addressbook/external.py]: I'm starting to believe in tests after all :) nice one.

File: tests/addressbook/external_test.py (hunk 2 above)
Comment (dcbaker) [inline tests/addressbook/external_test.py]: No, but it's being explict that this string is meant to be passed into one of the `re` functions. It's not the end of the world either way.

**Pull request merged**